cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -115,3 +117,120 @@ const char *sc_mount_opt2str(char *buf, size_t buf_size, unsigned long flags)
 	}
 	return buf;
 }
+
+const char *sc_mount_cmd(char *buf, size_t buf_size, const char *source, const char
+			 *target, const char *filesystemtype,
+			 unsigned long mountflags, const
+			 void *data)
+{
+	sc_string_init(buf, buf_size);
+	sc_string_append(buf, buf_size, "mount");
+
+	// Add filesysystem type if it's there and doesn't have the special value "none"
+	if (filesystemtype != NULL && strcmp(filesystemtype, "none") != 0) {
+		sc_string_append(buf, buf_size, " -t ");
+		sc_string_append(buf, buf_size, filesystemtype);
+	}
+	// Check for some special, dedicated syntax. Collect the flags that were
+	// displayed this way so that they are not repeated with -o foo syntax.
+	int used_special_flags = 0;
+
+	// Bind-ounts (bind)
+	if (mountflags & MS_BIND) {
+		if (mountflags & MS_REC) {
+			sc_string_append(buf, buf_size, " --rbind");
+		} else {
+			sc_string_append(buf, buf_size, " --bind");
+		}
+		used_special_flags |= MS_BIND | MS_REC;
```

Comment:
This is done so that we don't redundantly display `rec` in the generic `-o` section below. The idea is that since this only makes sense with one of the places that already handle `MS_REC` we don't display it in the generic section.